incentivization.md

```diff
@@ -52,231 +47,186 @@ Early P2P file-sharing networks employed reputation-based approaches and sticky
-If the server "pays first", it assumes risk.
-This risk should be decreased or paid for.
-Decreasing the risk means keeping track of the clients' reputation from the server's standpoint, which may endanger clients' privacy.
-Paying for the risk means increasing prices (i.e., well-behaved clients in aggregate pay for free-riders).
-We suggest that the preferable design is the opposite: the client assumes the risk.
-Why this is better:
-- it's more likely that the server is professionalized: serving data is its business which it wouldn't want to sabotage;
-- the client keeps their privacy, essentially paying for privacy with taking on more risk - this is OK, as risk is "anonymous", and reputation is not.
+If the server "pays first", it assumes risk, which should be decreased or paid for.
+Decreasing the risk means that the client keeps track of the clients' reputation, which endangers privacy.
+Paying for the risk means higher prices (well-behaved clients pay for free-riders).
 
-### Reputation accounting
+We propose that the client assumes the risk and pays for it because:
+- the server is more likely to be professionalized, so dropping paid requests would sabotage its business;
+- the client pays for their privacy by assuming risk, which is acceptable (risk is "anonymous", reputation is not).
 
-Our protocol assumes that the client trusts the server.
-In particular, the client pays first, and then hopes that the server sends back the response.
-A server may technically take the money and do nothing.
-To discourage this behavior, we use reputation: a client keeps track of the server's behavior.
+### Reputation accounting
 
-The MVP version could be:
-- all servers start with zero reputation
+We use reputation to discourage the server from taking the payment and not responding.
+The client keeps track of the server's reputation:
+- all servers start with zero reputation;
 - if the server honors the request, it gets +1;
-- if the server does not respond after the initial query, it gets -1;
-- if the server takes the money and _then_ does not respond, it gets banned (this client will never query it again).
+- if the server does not respond to the initial request, it gets -1;
+- if the server takes the money and does not respond before a timeout, the client will never query it again.
 
 Potential issues:
-- An attacker can establish new server identities and continually run away with clients' money.
-	- countermeasures:
-		- a client only queries "trusted" servers (centralization);
-		- when querying a new server, a client first sends a small (i.e. cheap) request to not risk too much.
-- Think about how the ban mechanism can be abused. Can an attacker "frame" competitors' servers so that many clients ban them?
-
-### Results cross-checking
+- An attacker can establish new server identities and continue running away with clients' money. Countermeasures:
+	- a client only queries "trusted" servers (which however leads to centralization);
```

Comment:
You can have a cap on reputation points (or a decaying function) such that at some stage you give more weight to "negative reputation" and newer nodes can catch up to the reputation of old, trustful but centralizing servers.

Reply:
Loosely speaking clients consistently retrieving files from servers are incentivized not to over concentrate reliance on any one point of failure, so they should be exploring other servers to discover those that are reliable as well.

Inherited or transitive reputation may help with this as well. If Client A interacts via messaging or otherwise with Client B, and Client B trusts Server C, Client A should reasonably trust Server C even if never having directly interacted with it before.

Reply:
I'm not sure the following answers any particular concern, but I'll bring up the Lightning Network again. Its architecture is very decentralized in theory, but in practice a new node must decide who to open a channel to - and this means real money on the line, not just data exchange. As a result, clients often chose one of top-capacity or top-degree nodes (this information is [public](https://1ml.com/)), or rely on centralized rankings ([example](https://fulmo.org/bos-score.html)).

What I'm getting at it is: perhaps, the best security model we can achieve here is of the type "you use a centralized provider but have the right to switch to another centralized provider at any time". Maybe this is OK for our security requirements?..

Reply:
Extracted ideas from this thread into #48 #49, marking this as resolved.